types/setting.go

```diff
@@ -1726,3 +1728,15 @@ func SetSettingDefinition(name SettingName, definition SettingDefinition) {
 	defer settingDefinitionsLock.Unlock()
 	settingDefinitions[name] = definition
 }
+
+func GetSettingForSystemManagedComponents() sets.Set[SettingName] {
+	settingList := sets.New[SettingName]()
+	for settingName, setting := range settingDefinitions {
+		if setting.Category == SettingCategoryDangerZone {
+			settingList = settingList.Insert(settingName)
+		}
+	}
+	settingList = settingList.Insert(SettingNameV1DataEngine, SettingNameV2DataEngine)
```

Comment:
Move `SettingNameV1DataEngine, SettingNameV2DataEngine` to danger zone category.

Reply:
Moved `SettingNameV1DataEngine, SettingNameV2DataEngine, and SettingNameV2DataEngineGuaranteedInstanceManagerCPU` to the danger zone category.
Thanks.